book/src/volume-cloning.md

```diff
@@ -0,0 +1,48 @@
+# Volume Cloning
+
+## Status and Releases
+
+** Status:** Alpha
+
+Compatible with CSI Version | Supported CSI external Provisioner Version | Min k8s Version | Max k8s version
+--|--|--|--|--|--
+v1.0  | v1.3 | v1.15 | -
+
+## Overview
+
+A Clone is defined as a duplicate of an existing Kubernetes Volume.  For more information on cloning in Kubernetes see the concepts doc for [Volume Cloning](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#volume-cloning).  A storage provider that allows volume cloning as a create feature, may choose to implement volume cloning via a control-plan CSI RPC call.
+
+For details regarding the kuberentes API for volume cloning, please see [kubernetes concepts](https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/).
+
+## Implementing Volume cloning functionality
+
+To implement volume cloning the CSI driver MUST:
+
+1. Implement checks for `csi.CreateVolumeRequest.VolumeContentSource` in the plugin's `CreateVolume` function implementation.
+2. Implement `CLONE_VOLUME` controller capability.
+
+Note that the `external-provisioner` side car currently handles the checks listed above, as well as verifying the specified source can be fetched using the Kubernetes API.
+
+It is the responsibility of the storage plugin to either implement an expansion after clone if a provision request size is greater than the source.
```

Comment:
Or can you let the external-resizer handle the resizing independently after clone?

Reply:
arg... yes, I'll add that as a possible option.